### ISSUE TITLE
Are images not extracted?

### ISSUE THREAD

**Issue by uneetkumarsingh** [facebookresearch/nougat#83]
>> nougat <PDF_file>

Only gives a mmd file and not images in the pdf. is there a way to get those too?

**Comment by marwinsteiner** [comment]
not currently supported. on the facebook research project page you can see some sample generations. images are not pulled out right now.

**Comment by lukas-blecher** [comment]
That's correct. I'll add this to the FAQ but here's a potential workaround if you need to get this working:

During the data generation phase, we processed all images with [`pdffigures2`](https://github.com/allenai/pdffigures2). Its a really useful tool for exactly what you're looking for: It recognizes all figures (and tables) + their caption and gives you the bounding boxes. From that you can extract the images of a PDF and since you also get the captions, you can match them with the captions transcribed by nougat (They will always appear at the end of a page).

**Comment by marwinsteiner** [comment]
> That's correct. I'll add this to the FAQ but here's a potential workaround if you need to get this working:
> 
> During the data generation phase, we processed all images with [`pdffigures2`](https://github.com/allenai/pdffigures2). Its a really useful tool for exactly what you're looking for: It recognizes all figures (and tables) + their caption and gives you the bounding boxes. From that you can extract the images of a PDF and since you also get the captions, you can match them with the captions transcribed by nougat (They will always appear at the end of a page).

@lukas-blecher do you do any layout detection right now? do you do layout detection with pdffigures2?

**Comment by lukas-blecher** [comment]
No, nougat works end to end without any layout detection

**Comment by marwinsteiner** [comment]
I'm talking about a domain-agnostic converter. A PDF to TEX converter for academic documents is nice. but as the frequently asked questions show, many people try using this technology (mostly unsuccessfully) outside the STEM subdomain. Clearly the need for a feature-agnostic converter is there, no? Perhaps end-to-end might be the wrong approach with an agnostic converter. Maybe you have to try detecting a layout before you can try recognizing text?

**Comment by lukas-blecher** [comment]
This is no longer related to what the issue is about. And also outside the scope of this research product.
You're right, it would be an interesting and immensely useful tool, and I hope that this work lowers the entry bar for such a project.

**Comment by sidharthrajaram** [comment]
> That's correct. I'll add this to the FAQ but here's a potential workaround if you need to get this working:
> 
> During the data generation phase, we processed all images with [`pdffigures2`](https://github.com/allenai/pdffigures2). Its a really useful tool for exactly what you're looking for: It recognizes all figures (and tables) + their caption and gives you the bounding boxes. From that you can extract the images of a PDF and since you also get the captions, you can match them with the captions transcribed by nougat (They will always appear at the end of a page).

@lukas-blecher I was trying to do something similar and was just curious about one thing, were you able to use `pdffigures2` programmatically in your python code or did you use a subprocess (or something like that) to directly use a JAR of it?

**Comment by Nicholas-Schaub** [comment]
https://github.com/facebookresearch/nougat/blob/47c77d70727558b4a2025005491ecb26ee97f523/nougat/dataset/pdffigures.py#L11
